SystemServices/SystemServices.cpp

```diff
@@ -3761,55 +3759,192 @@ namespace WPEFramework {
+                if(parameters["WAKEUPSRC_CEC"].Boolean())
+                    config |= 1<<WAKEUPSRC_CEC;
+                LOGWARN(" %s: %d WAKEUPSRC_CECsrcType:%x  config :%x \n",__FUNCTION__,__LINE__,srcType ,config);
+            }
+            if(parameters.HasLabel("WAKEUPSRC_LAN"))
+            {
+                srcType |= 1<<WAKEUPSRC_LAN;
+                if(parameters["WAKEUPSRC_LAN"].Boolean())
+                    config |= 1<<WAKEUPSRC_LAN;
+                LOGWARN(" %s: %d WAKEUPSRC_LANsrcType:%x  config :%x \n",__FUNCTION__,__LINE__,srcType ,config);
+            }
+            if(srcType) {
+                IARM_Bus_PWRMgr_WakeupSrcConfig_Param_t param;
+                param.pwrMode = powerState;
+                param.srcType = srcType;
+                param.config = config;
+                IARM_Result_t res = IARM_Bus_Call(IARM_BUS_PWRMGR_NAME,
+                                       IARM_BUS_PWRMGR_API_SetWakeupSrcConfig, (void *)&param,
+                                       sizeof(param));
+                if (IARM_RESULT_SUCCESS == res) {
+                    status = true;
+                } else {
+                    status = false;
                 }
+            }
+            returnResponse(status);
+        }
+        
+        /***
+         * @brief : To get the wakeup source configuration.
+         * @param1[out] : {"params":{ "wakeupSrc": <int>, "config": <int>}
+         * @param2[out] : {"result":{"success":<bool>}}
+         * @return     : Core::<StatusCode>
+         */
 
-                if(paramErr == 0) {
-
-                    IARM_Result_t res = IARM_Bus_Call(IARM_BUS_PWRMGR_NAME,
-                                           IARM_BUS_PWRMGR_API_SetWakeupSrcConfig, (void *)&param,
-                                           sizeof(param));
-
-                    if (IARM_RESULT_SUCCESS == res) {
-                        status = true;
-                    } else {
-                        status = false;
-                    }
+        uint32_t SystemServices::getWakeupSrcConfiguration(const JsonObject& parameters,
+                JsonObject& response)
+        {
+            IARM_Bus_PWRMgr_WakeupSrcConfig_Param_t param;
+            bool status = false;
+            IARM_Result_t res = IARM_Bus_Call(IARM_BUS_PWRMGR_NAME,
+                                  IARM_BUS_PWRMGR_API_GetWakeupSrcConfig, (void *)&param,
+                                  sizeof(param));
+            if (IARM_RESULT_SUCCESS == res) {
+               LOGWARN(" %s: %d res:%d srcType :%x  config :%x \n",__FUNCTION__,__LINE__,res,param.srcType,param.config);
+               status = true;
+               if( param.srcType & (1<<WAKEUPSRC_VOICE))
+               {
+                    response["WAKEUPSRC_VOICE"] = (param.config & (1<<WAKEUPSRC_VOICE))?true:false;
+                    LOGWARN(" %s: %d  WAKEUPSRC_VOICE value:%d  \n",__FUNCTION__,__LINE__,param.config & (1<<WAKEUPSRC_VOICE));
```

Comment:
no need to print here as combined value already printed in line 3878

Reply:
removed